wordpress-mobile/WordPress-iOS
# Issues/15266 fix preview url

Fixes #15266
Refs #14220 

This PR changes the way the app handles previews to resolve the issue reported in #15266 and potentially mitigates the issue reported in #15296 but needs confirmation. The changes include:
- Adding a convenience getter to the Blog model for checking if a site has a mapped domain.
- Updates the PreviewNonceHandler to only add the preview argument when viewing non published posts, or published post that have remote changes, and to only unmap a mapped URL in these cases.
- Updates the RequestAuthenticator to explicitly proxy requests through r-login for mapped domains, excepted when loading an unmapped URL.   This change was necessary to view published and drafts posts consistently on sites marked "coming soon" that have mapped domains.  I suspect this is the change that resolves 15296.

To test:
Perform the following tests
- View a published post. 
- Make a change to a draft post and preview the change. 
- Load View Site and confirm you are logged in.

Using the following site configurations:
- a simple wordpress.com site. (private and public)
- a simple wordpress.com site with a custom domain. (private and public)
- an atomic site
- a self-hosted + jetpack site.

Ping me if you need a site to test with.  I have examples of them all :)

There's a lot to cover with these checks so maybe a couple of folks could help out.  @guarani , @diegoreymendez would y'all be able to take a peek? 

PR submission checklist:
